src/traces/scatter/attributes.js

```diff
@@ -205,7 +205,7 @@ module.exports = {
         ].join(' ')
     },
     hovertemplate: hovertemplateAttrs({}, {
-        keys: ['marker.size', 'marker.color']
+        keys: ['marker.size']
```

Comment:
Why is `marker.color` gone? In fact, all `arrayOk` attributes should be part of this list. They get added to the event data via 

https://github.com/plotly/plotly.js/blob/2ceb5c3e7981cc850727f7406f6677c4d74be044/src/components/fx/helpers.js#L161-L186

Now, maybe we don't need to hardcode all the `arrayOk` attributes into `keys` here, maybe we could just mention that all "per-point" values are available for `hovertemplate`.

Reply:
`marker.color` is gone because it is not emitted in the event data. It used to work because `hovertemplateString` would be called with the full `trace` object as one of its argument but I removed this in 43a4cd9 to make sure everything is on par with event data. Therefore, if we want `marker.color`, I need to add it to the event data for `scatter`.

Reply:
> `marker.color` is gone because it is not emitted in the event data

Try

```js
Plotly.newPlot(gd, [{
  y: [1, 2, 3],
  marker: {color: [1, 2, 3]}
}])
gd.on('plotly_hover', console.log)
```

Reply:
It works if it's an array but not if it isn't? :thinking: That's why the test was failing: the mock I was using doesn't specify `marker.color` as an array.
```
Plotly.newPlot(gd, [{
  y: [1, 2, 3],
  marker: {color: 'blue'}
}])
gd.on('plotly_hover', console.log)
```

Reply:
> t works if it's an array but not if it isn't?

yes, that's the expected behaviour.

Reply:
Ok so I added it and now run the test with a mock that specifies marker size and color as arrays. And it now passes.

Now for extra points, I just need to add those extra event data keys in `constants.js`.

Reply:
Done in 583eb97

Reply:
From that same https://github.com/plotly/plotly.js/pull/3126#discussion_r233487700

> Now, maybe we don't need to hardcode all the `arrayOk` attributes into `keys` here,

I think it might be best to **NOT** hardcoded all `arrayOk` attributes into that `keys` list, as it will be hard to remember to append it everytime we add another `arrayOk` attribute. We should keep that `keys` list only for keys that have no corresponding schema attributes e.g `binNumber` in histogram traces, **and** we should mention that all `arrayOk` (aka per-point) attribute are available to `hovertemplate` in the `components/fx/hovertemplate_attributes'` description.

Sorry for not been clearer.